src/phpDocumentor/Phpdoc2XmlToArrayConverter.php

```diff
@@ -0,0 +1,154 @@
+            $defaultPackageName = ((string) $phpDocumentor->parser->{'default-package-name'}) ?: $defaultPackageName;
+            $template           = ((string) $phpDocumentor->transformations->template->attributes()->name) ?: $template;
+
+            if (isset($phpDocumentor->parser->files)) {
+                if (isset($phpDocumentor->parser->files->{'ignore-hidden'})) {
+                    $ignoreHidden = filter_var($phpDocumentor->parser->files->{'ignore-hidden'}, FILTER_VALIDATE_BOOLEAN);
+                }
+
+                if (isset($phpDocumentor->parser->files->{'ignore-symlinks'})) {
+                    $ignoreSymlinks = filter_var($phpDocumentor->parser->files->{'ignore-symlinks'}, FILTER_VALIDATE_BOOLEAN);
+                }
+            }
+        }
+
+        $outputDirectory = ((string) $phpDocumentor->parser->target) ?: 'file://build/docs';
+        $sourcePath      = ((string) $phpDocumentor->parser->files->directory) ?: 'src';
+
+        $phpdoc2Array = [
+            'phpdocumentor' => [
+                'paths'     => [
+                    'output' => (string) (new Dsn($outputDirectory))->getPath(),
+                    'cache'  => '/tmp/phpdoc-doc-cache'
+                ],
+                'versions'  => [
+                    '1.0.0' => [
+                        'folder' => '',
+                        'api'    => [
+                            'format'               => 'php',
+                            'source'               => [
+                                'dsn'   => 'file://.',
+                                'paths' => [
+                                    0 => (string) (new Dsn($sourcePath))->getPath(),
+                                ]
+                            ],
+                            'ignore'               => [
+                                'hidden'   => $ignoreHidden,
+                                'symlinks' => $ignoreSymlinks,
+                                'paths'    => [
+                                    0 => 'src/ServiceDefinitions.php'
+                                ]
+                            ],
+                            'extensions'           => $extensions,
+                            'visibility'           => $visibility,
+                            'default-package-name' => $defaultPackageName,
+                            'markers'              => $markers,
+                        ],
+                        'guide'  => [
+                            'format' => 'rst',
+                            'source' => [
+                                'dsn'   => 'file://../phpDocumentor/phpDocumentor2',
+                                'paths' => [
+                                    0 => 'docs'
+                                ]
+                            ]
+                        ]
+                    ]
+                ],
+                'templates' => [
+                    0 => [
```

Comment:
You can pass multiple template entries in the phpDocumentor 2 XML; these should all be added here